### ISSUE TITLE
[runner-label] should support larger runners

### ISSUE THREAD

**Issue by rvandegrift** [rhysd/actionlint#337]
Github allows larger runners to be used with labels like `ubuntu-latest-4-cores`.  The `runner-label` check 
For details: https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners

Example failure: https://rhysd.github.io/actionlint/#eJxtjsEKwkAMRO/9ijl4zVKkB9m/2ZZgFd2UTQJK6b/bpS168BQeTOaN5IjJdWzu0mtsAGO1egG1koyv742AZ7Jyex0EFM+Zy5cBgveezemRag11NEhh/Zc4t6Ht6PITWOuUZJ1zmufdFTYFlmUfxJMePqoPETyMghDCB+pjN+w=

**Comment by ClementWalter** [comment]
> You can add `actionlint.yaml` under `.github` folder with this:
> 
> ```yaml
> self-hosted-runner:
>   labels:
>     - ubuntu-latest-4-cores
>     - ubuntu-latest-8-cores
> ```

this is a patch, because they are not "self hosted" but official large GH runners

**Comment by ericcrosson-bitgo** [comment]
I believe this has been closed by https://github.com/rhysd/actionlint/commit/64902ec49b95dbd4e88bf2f543885edb18bbff09

**Comment by pbrisbin** [comment]
FYI, it's possible to create GitHub-hosted runners with any name. For example, we have one named `ubuntu-22.04-4-cores`, which we then added to `self-hosted-runner` despite not being "self hosted". I wonder if that configuration key could just be renamed* to `custom-runner`?

*And by that I mean adding a new key and deprecating old.
